review/Dockerfile

```diff
@@ -1,14 +1,17 @@
 # build stage
-FROM golang:latest AS builder
+FROM golang:1.23 AS builder
 
-RUN apk add git gcc libc-dev
+WORKDIR /usr/src/app
 
-WORKDIR /go/src/coupon-service
+COPY go.mod go.sum ./
+RUN go mod download && go mod verify
 
 COPY . .
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -v -o /usr/local/bin/app cmd/coupon_service/*.go
 
-WORKDIR /go/src/coupon-service/cmd/export
-RUN go build -o main .
+FROM scratch
```

Comment:
I used here `FROM scratch` because It's security best practice. We don't have shell env in such image, so no one can connect the shell in container and read secrets or access any storages from container. On the other hand we can't use tools inside container for debug. As alternative we can use Alpine images.